src/gui/transferlistwidget.h

```diff
@@ -33,16 +33,15 @@
 #include <QtContainerFwd>
 #include <QTreeView>
 
-#include "base/bittorrent/infohash.h"
```

Comment:
I get compile errors because of this: 
<details>
  <summary>Click</summary>
  
```
In file included from /usr/include/c++/11.2.0/bits/move.h:57,
                 from /usr/include/c++/11.2.0/bits/stl_pair.h:59,
                 from /usr/include/c++/11.2.0/bits/stl_algobase.h:64,
                 from /usr/include/c++/11.2.0/memory:63,
                 from moc_transferlistwidget.cpp:9:
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘struct std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>’:
/usr/include/c++/11.2.0/type_traits:115:12:   required from ‘struct std::__or_<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> >’
/usr/include/c++/11.2.0/type_traits:167:12:   required from ‘struct std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> >’
/usr/include/c++/11.2.0/type_traits:133:12:   required from ‘struct std::__and_<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:162:12:   required from ‘struct std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:177:64:   required from ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/qt6/QtCore/qdebug.h:248:27:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/qt6/QtCore/qtypeinfo.h:375:89:   required by substitution of ‘template<class Stream, class T> struct QTypeTraits::has_ostream_operator<Stream, T, std::void_t<decltype ((reference<Stream>() << const_reference<T>()))> > [with Stream = QDebug; T = QSet<BitTorrent::TorrentID>]’
/usr/include/qt6/QtCore/qtypeinfo.h:378:81:   required from ‘constexpr const bool QTypeTraits::has_ostream_operator_v<QDebug, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2212:77:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:1372:38: error: invalid use of incomplete type ‘class BitTorrent::TorrentID’
 1372 |     : public integral_constant<bool, __is_base_of(_Base, _Derived)>
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from moc_transferlistwidget.cpp:10:
gui/transferlistwidget.h:46:11: note: forward declaration of ‘class BitTorrent::TorrentID’
   46 |     class TorrentID;
      |           ^~~~~~~~~
In file included from /usr/include/c++/11.2.0/bits/move.h:57,
                 from /usr/include/c++/11.2.0/bits/stl_pair.h:59,
                 from /usr/include/c++/11.2.0/bits/stl_algobase.h:64,
                 from /usr/include/c++/11.2.0/memory:63,
                 from moc_transferlistwidget.cpp:9:
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘struct std::__or_<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> >’:
/usr/include/c++/11.2.0/type_traits:167:12:   required from ‘struct std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> >’
/usr/include/c++/11.2.0/type_traits:133:12:   required from ‘struct std::__and_<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:162:12:   required from ‘struct std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:177:64:   required from ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
/usr/include/qt6/QtCore/qdebug.h:248:27:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/qt6/QtCore/qtypeinfo.h:375:89:   required by substitution of ‘template<class Stream, class T> struct QTypeTraits::has_ostream_operator<Stream, T, std::void_t<decltype ((reference<Stream>() << const_reference<T>()))> > [with Stream = QDebug; T = QSet<BitTorrent::TorrentID>]’
/usr/include/qt6/QtCore/qtypeinfo.h:378:81:   required from ‘constexpr const bool QTypeTraits::has_ostream_operator_v<QDebug, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2212:77:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:115:12: error: ‘value’ is not a member of ‘std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>’
  115 |     struct __or_<_B1, _B2>
      |            ^~~~~~~~~~~~~~~
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’:
/usr/include/qt6/QtCore/qdebug.h:248:27:   required by substitution of ‘template<class Container, class ... T> using QDebugIfHasDebugStreamContainer = std::enable_if_t<conjunction_v<std::disjunction<std::is_base_of<Container, T>, QTypeTraits::has_ostream_operator<QDebug, T, void> >...>, QDebug> [with Container = QSet<BitTorrent::TorrentID>; T = {BitTorrent::TorrentID}]’
/usr/include/qt6/QtCore/qdebug.h:315:52:   required by substitution of ‘template<class T> QDebugIfHasDebugStreamContainer<QSet<T>, T> operator<<(QDebug, const QSet<T>&) [with T = BitTorrent::TorrentID]’
/usr/include/qt6/QtCore/qtypeinfo.h:375:89:   required by substitution of ‘template<class Stream, class T> struct QTypeTraits::has_ostream_operator<Stream, T, std::void_t<decltype ((reference<Stream>() << const_reference<T>()))> > [with Stream = QDebug; T = QSet<BitTorrent::TorrentID>]’
/usr/include/qt6/QtCore/qtypeinfo.h:378:81:   required from ‘constexpr const bool QTypeTraits::has_ostream_operator_v<QDebug, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2212:77:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:177:64: error: ‘value’ is not a member of ‘std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDebug, BitTorrent::TorrentID, void> > >’
  177 |     inline constexpr bool conjunction_v = conjunction<_Bn...>::value;
      |                                                                ^~~~~
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘struct std::__or_<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> >’:
/usr/include/c++/11.2.0/type_traits:167:12:   required from ‘struct std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> >’
/usr/include/c++/11.2.0/type_traits:133:12:   required from ‘struct std::__and_<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:162:12:   required from ‘struct std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:177:64:   required from ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/qt6/QtCore/qdatastream.h:356:27:   required by substitution of ‘template<class Container, class ... T> using QDataStreamIfHasOStreamOperatorsContainer = std::enable_if_t<conjunction_v<std::disjunction<std::is_base_of<Container, T>, QTypeTraits::has_ostream_operator<QDataStream, T, void> >...>, QDataStream&> [with Container = QSet<BitTorrent::TorrentID>; T = {BitTorrent::TorrentID}]’
/usr/include/qt6/QtCore/qdatastream.h:455:62:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/qt6/QtCore/qtypeinfo.h:378:81:   required from ‘constexpr const bool QTypeTraits::has_ostream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qtypeinfo.h:394:47:   required from ‘constexpr const bool QTypeTraits::has_stream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2325:60:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:115:12: error: ‘value’ is not a member of ‘std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>’
  115 |     struct __or_<_B1, _B2>
      |            ^~~~~~~~~~~~~~~
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> > >’:
/usr/include/qt6/QtCore/qdatastream.h:356:27:   required by substitution of ‘template<class Container, class ... T> using QDataStreamIfHasOStreamOperatorsContainer = std::enable_if_t<conjunction_v<std::disjunction<std::is_base_of<Container, T>, QTypeTraits::has_ostream_operator<QDataStream, T, void> >...>, QDataStream&> [with Container = QSet<BitTorrent::TorrentID>; T = {BitTorrent::TorrentID}]’
/usr/include/qt6/QtCore/qdatastream.h:455:62:   required by substitution of ‘template<class T> QDataStreamIfHasOStreamOperatorsContainer<QSet<T>, T> operator<<(QDataStream&, const QSet<T>&) [with T = BitTorrent::TorrentID]’
/usr/include/qt6/QtCore/qtypeinfo.h:375:89:   required by substitution of ‘template<class Stream, class T> struct QTypeTraits::has_ostream_operator<Stream, T, std::void_t<decltype ((reference<Stream>() << const_reference<T>()))> > [with Stream = QDataStream; T = QSet<BitTorrent::TorrentID>]’
/usr/include/qt6/QtCore/qtypeinfo.h:378:81:   required from ‘constexpr const bool QTypeTraits::has_ostream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qtypeinfo.h:394:47:   required from ‘constexpr const bool QTypeTraits::has_stream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2325:60:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:177:64: error: ‘value’ is not a member of ‘std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_ostream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
  177 |     inline constexpr bool conjunction_v = conjunction<_Bn...>::value;
      |                                                                ^~~~~
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘struct std::__or_<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> >’:
/usr/include/c++/11.2.0/type_traits:167:12:   required from ‘struct std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> >’
/usr/include/c++/11.2.0/type_traits:133:12:   required from ‘struct std::__and_<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:162:12:   required from ‘struct std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/c++/11.2.0/type_traits:177:64:   required from ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
/usr/include/qt6/QtCore/qdatastream.h:363:27:   required by substitution of ‘template<class Container, class ... T> using QDataStreamIfHasIStreamOperatorsContainer = std::enable_if_t<conjunction_v<std::disjunction<std::is_base_of<Container, T>, QTypeTraits::has_istream_operator<QDataStream, T, void> >...>, QDataStream&> [with Container = QSet<BitTorrent::TorrentID>; T = {BitTorrent::TorrentID}]’
/usr/include/qt6/QtCore/qdatastream.h:449:62:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/usr/include/qt6/QtCore/qtypeinfo.h:389:81:   required from ‘constexpr const bool QTypeTraits::has_istream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qtypeinfo.h:394:84:   required from ‘constexpr const bool QTypeTraits::has_stream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2325:60:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:115:12: error: ‘value’ is not a member of ‘std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>’
  115 |     struct __or_<_B1, _B2>
      |            ^~~~~~~~~~~~~~~
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘constexpr const bool std::conjunction_v<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> > >’:
/usr/include/qt6/QtCore/qdatastream.h:363:27:   required by substitution of ‘template<class Container, class ... T> using QDataStreamIfHasIStreamOperatorsContainer = std::enable_if_t<conjunction_v<std::disjunction<std::is_base_of<Container, T>, QTypeTraits::has_istream_operator<QDataStream, T, void> >...>, QDataStream&> [with Container = QSet<BitTorrent::TorrentID>; T = {BitTorrent::TorrentID}]’
/usr/include/qt6/QtCore/qdatastream.h:449:62:   required by substitution of ‘template<class T> QDataStreamIfHasIStreamOperatorsContainer<QSet<T>, T> operator>>(QDataStream&, QSet<T>&) [with T = BitTorrent::TorrentID]’
/usr/include/qt6/QtCore/qtypeinfo.h:386:89:   required by substitution of ‘template<class Stream, class T> struct QTypeTraits::has_istream_operator<Stream, T, std::void_t<decltype ((reference<Stream>() >> reference<T>()))> > [with Stream = QDataStream; T = QSet<BitTorrent::TorrentID>]’
/usr/include/qt6/QtCore/qtypeinfo.h:389:81:   required from ‘constexpr const bool QTypeTraits::has_istream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qtypeinfo.h:394:84:   required from ‘constexpr const bool QTypeTraits::has_stream_operator_v<QDataStream, QSet<BitTorrent::TorrentID> >’
/usr/include/qt6/QtCore/qmetatype.h:2325:60:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:177:64: error: ‘value’ is not a member of ‘std::conjunction<std::disjunction<std::is_base_of<QSet<BitTorrent::TorrentID>, BitTorrent::TorrentID>, QTypeTraits::has_istream_operator<QDataStream, BitTorrent::TorrentID, void> > >’
  177 |     inline constexpr bool conjunction_v = conjunction<_Bn...>::value;
      |                                                                ^~~~~
In file included from /usr/include/qt6/QtCore/qabstractitemmodel.h:44,
                 from /usr/include/qt6/QtWidgets/qabstractitemview.h:45,
                 from /usr/include/qt6/QtWidgets/qtreeview.h:44,
                 from /usr/include/qt6/QtWidgets/QTreeView:1,
                 from gui/transferlistwidget.h:34,
                 from moc_transferlistwidget.cpp:10:
/usr/include/qt6/QtCore/qhash.h: In instantiation of ‘struct QHashPrivate::Node<BitTorrent::TorrentID, QHashDummyValue>’:
/usr/include/qt6/QtCore/qhash.h:462:11:   required from ‘struct QHashPrivate::Data<QHashPrivate::Node<BitTorrent::TorrentID, QHashDummyValue> >’
/usr/include/qt6/QtCore/qhash.h:776:16:   required from ‘QHash<Key, T>::QHash(const QHash<Key, T>&) [with Key = BitTorrent::TorrentID; T = QHashDummyValue]’
/usr/include/qt6/QtCore/qset.h:53:7:   required from ‘static constexpr void (* QtPrivate::QMetaTypeForType<S>::getCopyCtr())(const QtPrivate::QMetaTypeInterface*, void*, const void*) [with S = QSet<BitTorrent::TorrentID>; QtPrivate::QMetaTypeInterface::CopyCtrFn = void (*)(const QtPrivate::QMetaTypeInterface*, void*, const void*)]’
/usr/include/qt6/QtCore/qmetatype.h:2319:54:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/qt6/QtCore/qhash.h:156:9: error: ‘QHashPrivate::Node<Key, QHashDummyValue>::key’ has incomplete type
  156 |     Key key;
      |         ^~~
In file included from moc_transferlistwidget.cpp:10:
gui/transferlistwidget.h:46:11: note: forward declaration of ‘class BitTorrent::TorrentID’
   46 |     class TorrentID;
      |           ^~~~~~~~~
In file included from /usr/include/c++/11.2.0/bits/move.h:57,
                 from /usr/include/c++/11.2.0/bits/stl_pair.h:59,
                 from /usr/include/c++/11.2.0/bits/stl_algobase.h:64,
                 from /usr/include/c++/11.2.0/memory:63,
                 from moc_transferlistwidget.cpp:9:
/usr/include/c++/11.2.0/type_traits: In instantiation of ‘struct std::is_nothrow_destructible<BitTorrent::TorrentID>’:
/usr/include/c++/11.2.0/type_traits:3166:35:   required from ‘constexpr const bool std::is_nothrow_destructible_v<BitTorrent::TorrentID>’
/usr/include/qt6/QtCore/qhash.h:780:28:   required from ‘QHash<Key, T>::~QHash() [with Key = BitTorrent::TorrentID; T = QHashDummyValue]’
/usr/include/qt6/QtCore/qset.h:53:7:   required from ‘static constexpr void (* QtPrivate::QMetaTypeForType<S>::getCopyCtr())(const QtPrivate::QMetaTypeInterface*, void*, const void*) [with S = QSet<BitTorrent::TorrentID>; QtPrivate::QMetaTypeInterface::CopyCtrFn = void (*)(const QtPrivate::QMetaTypeInterface*, void*, const void*)]’
/usr/include/qt6/QtCore/qmetatype.h:2319:54:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/c++/11.2.0/type_traits:900:52: error: static assertion failed: template argument must be a complete class or an unbounded array
  900 |       static_assert(std::__is_complete_or_unbounded(__type_identity<_Tp>{}),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/11.2.0/type_traits:900:52: note: ‘std::__is_complete_or_unbounded<std::__type_identity<BitTorrent::TorrentID> >((std::__type_identity<BitTorrent::TorrentID>{}, std::__type_identity<BitTorrent::TorrentID>()))’ evaluates to false
In file included from /usr/include/qt6/QtCore/qabstractitemmodel.h:44,
                 from /usr/include/qt6/QtWidgets/qabstractitemview.h:45,
                 from /usr/include/qt6/QtWidgets/qtreeview.h:44,
                 from /usr/include/qt6/QtWidgets/QTreeView:1,
                 from gui/transferlistwidget.h:34,
                 from moc_transferlistwidget.cpp:10:
/usr/include/qt6/QtCore/qhash.h: In instantiation of ‘QHash<Key, T>::~QHash() [with Key = BitTorrent::TorrentID; T = QHashDummyValue]’:
/usr/include/qt6/QtCore/qset.h:53:7:   required from ‘static constexpr void (* QtPrivate::QMetaTypeForType<S>::getCopyCtr())(const QtPrivate::QMetaTypeInterface*, void*, const void*) [with S = QSet<BitTorrent::TorrentID>; QtPrivate::QMetaTypeInterface::CopyCtrFn = void (*)(const QtPrivate::QMetaTypeInterface*, void*, const void*)]’
/usr/include/qt6/QtCore/qmetatype.h:2319:54:   required from ‘constexpr const QtPrivate::QMetaTypeInterface QtPrivate::QMetaTypeInterfaceWrapper<QSet<BitTorrent::TorrentID> >::metaType’
/usr/include/qt6/QtCore/qmetatype.h:2437:16:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* QtPrivate::qTryMetaTypeInterfaceForType() [with Unique = qt_meta_stringdata_TransferListWidget_t; TypeCompletePair = QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >]’
/usr/include/qt6/QtCore/qmetatype.h:2481:55:   required from ‘constexpr const QtPrivate::QMetaTypeInterface* const qt_incomplete_metaTypeArray [69]<qt_meta_stringdata_TransferListWidget_t, QtPrivate::TypeAndForceComplete<TransferListWidget, std::integral_constant<bool, true> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<BitTorrent::Torrent* const, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<CopyInfohashPolicy, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<int, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QString&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QSet<BitTorrent::TorrentID>&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const Path&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<const QModelIndex&, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<bool, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> >, QtPrivate::TypeAndForceComplete<void, std::integral_constant<bool, false> > >’
moc_transferlistwidget.cpp:350:1:   required from here
/usr/include/qt6/QtCore/qhash.h:780:28: error: static assertion failed: Types with throwing destructors are not supported in Qt containers.
  780 |         static_assert(std::is_nothrow_destructible_v<Key>, "Types with throwing destructors are not supported in Qt containers.");
      |                       ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/qt6/QtCore/qhash.h:780:28: note: ‘std::is_nothrow_destructible_v<BitTorrent::TorrentID>’ evaluates to false
```
</details>

In the end it says that my kit has "configuration issues which might be the root cause for this problem." so ignore this if you think it's my problem.

Reply:
I pushed another version, please try it out.
If it still errors, what is your qt version? cmake or qmake? and reverting just this line will fix it for you?

Reply:
Same errors.
Qt 6.2.3 qmake. cmake compiles fine and yes reverting this line fixes it for me for qmake.

Reply:
Reverted.